### ISSUE TITLE
autotools build with win_bash=True failing due to spaces in path

### ISSUE THREAD

**Issue by weatherhead99** [conan-io/conan#2344]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Hi,
I think the issue I'm having is best seen from this appveyor build fail:
https://ci.appveyor.com/project/weatherhead99/conan-gsl/build/job/83cy22h92yhwgseg

in my recipe I have
`abe = AutoToolsBuildEnvironment(self,win_bash=True)
abe.configure(configure_dir=os.path.join(self.source_folder,
                                                 "gsl-%s" % self.version),
                      args=["--enable-silent-rules"])`

conan 1.0.2, windows ?? (Appveyor)

**Comment by memsharded** [comment]
The relevant lines are:

```bash
[00:00:41] gsl/2.4@weatherhead99/testing: run_in_windows_bash: "C:\Program Files\Git\usr\bin\bash.exe" --login -c ^"cd \^"/c/users/appveyor/.conan/data/gsl/2.4/weatherhead99/testing/build/a09c0c1543526b525b054aeb2fb7e65dad25b2ce\^" ^&^& C:\Users\appveyor\.conan\data\gsl\2.4\weatherhead99\testing\build\a09c0c1543526b525b054aeb2fb7e65dad25b2ce\gsl-2.4/configure --enable-silent-rules --build=x86_64-w64-mingw32 --host=i686-w64-mingw32 ^"
[00:00:41] 'C:\Program' is not recognized as an internal or external command,
```

So Windows is not processing that even with quotes?? That is a new weirdness from Win.

**Comment by memsharded** [comment]
I have been able to reproduce it locally

Just a few comments while reviewing your recipe:

```python
self.cpp_info.libs = ["libgsl.lib", "libgslcblas.lib"]
if self.options["shared"] == True:
      self.cpp_info.libs.extend(["libgsl.dll", "libgslcblas.dll"])
```
Typically you don't need to add the dlls to ``cpp_info.libs``. Linking with the libs is more than enough. Also, typically you don't need to specify the extension. Unless I am missing something you can do:

```python
self.cpp_info.libs = ["libgsl", "libgslcblas"]
```

No need to compare to True, or access by [ ] you can just do:
```python
if self.options.shared:
    # something
```

You can do oneliners often, like e.g.:

````python
abe = AutoToolsBuildEnvironment(self,win_bash=(self.settings.os == "Windows"))
```
I hope they help, feel free to ask, I might be wrong or misunderstood some.

**Comment by memsharded** [comment]
In my case, the creation is failing for another reason, not because of spaces, because I don't have bash in a folder with spaces in my computer:

```bash
gsl/2.4@user/testing: run_in_windows_bash: C:\cmder\vendor\git-for-windows\bin\bash.exe --login -c ^"cd \^"/c/users/memsharded/.conan/data/gsl/2.4/user/testing/build/88104b9bd47d4d4ddc10097ecf36ba575ac994e2\^" ^&^& C:\Users\memsharded\.conan\data\gsl\2.4\user\testing\build\88104b9bd47d4d4ddc10097ecf36ba575ac994e2\gsl-2.4/configure --enable-silent-rules  ^"
/usr/bin/bash: C:Usersmemsharded.conandatagsl2.4usertestingbuild88104b9bd47d4d4ddc10097ecf36ba575ac994e2gsl-2.4/configure: No such file or directory
```
It seems the backslash are the cause.

**Comment by weatherhead99** [comment]
yes, actually this  isn't a bug I think.

The reason is that this recipe only builds on msys currently (i.e. not mingw or visual studio). I think there isn't an msys configuration installed on the appveyor instance, so you can't build packages that only work with msys.

**Comment by memsharded** [comment]
Released in 1.0.3, please update and report if necessary.
